lib/src/native/realm_core.dart

```diff
@@ -2995,7 +2995,15 @@ void _intoRealmQueryArg(Object? value, Pointer<realm_query_arg_t> realm_query_ar
     realm_query_arg.ref.arg = allocator<realm_value_t>();
     realm_query_arg.ref.nb_args = 1;
     realm_query_arg.ref.is_list = false;
-    _intoRealmValue(value, realm_query_arg.ref.arg.ref, allocator);
+    _intoRealmValueHack(value, realm_query_arg.ref.arg.ref, allocator);
+  }
+}
+
+void _intoRealmValueHack(Object? value, realm_value realm_value, Allocator allocator) {
+  if (value is GeoShape) {
+    _intoRealmValue(value.toString(), realm_value, allocator);
```

Comment:
Relying on `value.toString()` may cause some potential bugs. Imagine that the users extend the geo classes and override the `toString()`. 
I'm wondering 
- either we can add `toQuery` abstract method to `GeoShaps` that to be dedicated for the queries
- or to encapsulate this logic inside realm_core and implement a converter to query arguments, so we can be sure that the users won't override it.
I would rather prefer the second option, even though it is not aligned to OOP.

Reply:
GeoShape is sealed. Users cannot extend these classes.

Reply:
Discussed and decided no to allow users to extend these classes. All the classes that implement GeoShape should be marked as `final`.